cmd/serverless/main.go

```diff
@@ -276,16 +276,21 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 
 	// start appsec
 	var (
-		appsecSubProcessor   *httpsec.InvocationSubProcessor
+		appsecSubProcessor   invocationlifecycle.InvocationSubProcessor
 		appsecProxyProcessor *httpsec.ProxyLifecycleProcessor
 	)
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		appsecSubProcessor, appsecProxyProcessor, err = appsec.New()
+		subProcessor, proxySubProcessor, err := appsec.New()
 		if err != nil {
 			log.Error("appsec: could not start: ", err)
 		}
+		if subProcessor != nil {
+			appsecSubProcessor = subProcessor
+		} else if proxySubProcessor != nil {
```

Comment:
Are both `appsecSubProcessor` and `appsecProxyProcessor` supposed to be set? This `else if` will only get called if `subProcessor` is not equal to nil and it wasn't clear if that was intended. I'm guessing we use either a subprocessor or proxy, but not both?

Reply:
Yes, you got this right: appsec is either a "proxy subprocessor", or a "regular subprocessor".